File: plugins/kotlin/j2k/new/tests/testData/newJ2k/detectProperties/SetterTypeNotMatchInInterface.java
interface I {
    int getSomething();
    void setSomething(String value);
}

class C implements I {

    @Override
    public int getSomething() {
        return 0;
    }

    @Override
    public void setSomething(String value) {
        System.out.println("set");
    }

    public static void test(I i) {
        System.out.println(i.getSomething());
        i.setSomething("new");
    }
}

File: plugins/kotlin/j2k/shared/tests/testData/newJ2k/detectProperties/SetterTypeNotMatchInInterface.kt
internal interface I {
    val something: Int

    fun setSomething(value: String?)
}

internal class C : I {
    override val something: Int
        get() = 0

    override fun setSomething(value: String?) {
        println("set")
    }

    companion object {
        fun test(i: I) {
            println(i.something)
            i.setSomething("new")
        }
    }
}
